designcompose/src/main/java/com/android/designcompose/Utils.kt

```diff
@@ -753,6 +753,21 @@ internal fun ViewStyle.withExternalLayoutData(data: ExternalLayoutData): ViewSty
     return overrideStyle.build()
 }
 
+// Get the raw width in a view style from the width property if it is a fixed size, or from the
+// node_size property if not.
+internal fun ViewStyle.fixedWidth(density: Float): Float {
```

Comment:
These two methods seem a bit weird in sometimes considering density and sometimes not.

I'm not sure where node_style comes from, but if it is out of the DCF/DCB then it doesn't know about density, right?

Can you either move these back into the calculation for rotation of a dial, or into a file where we put the other rotation tricks?

Reply:
That's a good point, I copied the code in these functions so it might be an existing bug. I should multiply the node_size by density also. And yes node_size comes from the dcf file, and originally came from Figma as the raw size of the node.

Reply:
I got the code from `FrameRender.kt` and wanted a common function for it. It didn't make sense to keep it in FrameRender since the animation code was using it. I guess I could put it in `MathUtils.kt` but it doesn't seem like a math function -- it just retrieves data from `ViewStyle` and most other functions that do that are in `Utils.kt`.

Reply:
Ok - I guess I was thinking if we could find a way to do without knowing the pre-determined size then it would make sense to hide these methods so we don't use them in more places. I'm not sure how to get there given the other complexities in rotation, so we may as well have one implementation and make sure it's fully tested etc :)